doc/slit_tracing.rst

```diff
@@ -0,0 +1,287 @@
+
+Presently, we recommend that you err on the conservative
+side regarding thresholds, i.e. higher values of sigdetect,
+unless you have especially faint trace flat frames.
+
+Slit Gaps
+---------
+
+THIS METHOD IS NOT WELL TESTED NOR RECOMMENDED
+AT THIS STAGE (JXP).
+
+In cases where the trace frame contains slits that
+are uniformly illuminated in the spectral direction,
+and there is at least 5-10 pixels between the slits,
+the slit tracing algorithm generally works well.
+
+In the event
+that the slits are not uniformly illuminated, or if
+neighbouring slits are a little close (perhaps with
+some crosstalk), you may need to specify the slit gap
+using the argument::
+
+    trace slits maxgap 10
+
+in the event that the gap between all neighbouring slits is
+less than 10 pixels. This variable should not be used unless
+there is some crosstalk between slits, or in the event
+of close slits with a non-uniform illumination pattern.
+
+.. _trace-slit-user-defined:
+
+Slit Profile
+============
+
+With relatively short slits (often the case with
+multiobject or echelle data), the sky background
+is determined from relatively few pixels towards
+the edge of the slit, where the flux from a uniformly
+illuminated slit tends to roll off. To correct for
+this effect, PYPIT models the spatial slit profile
+of a trace frame (i.e. a flatfield with the same
+slit length as the science slit). The relevant set
+of parameters that determine the fit properties
+are given by::
+
+    reduce slitprofile perform False
+    reduce flatfield method bspline
+    reduce flatfield params [n]
+
+where n in the last line should be an integer or
+floating point number.
+
+The default setting is to not calculate the slit profile.
+To turn on this functionality, the argument of the
+first line above can be set to True. If the calculation
+is performed, the second line sets the method that should
+be used to determine the spatial slit profile.
+
+At this stage, PYPIT only supports the value 'bspline', where
+the knot spacing is set by the third line above. If the
+argument of reduce flatfield params is n >= 1, PYPIT
+will place a knot at every n pixels. Otherwise, if n < 1,
+PYPIT will place a knot at every k pixels, where k=n*N
+and N is the total number of pixels in the spectral
+direction. The number of knots in the spatial
+direction is set automatically by PYPIT, to be twice
+the number of pixels along the slit. Thus, the user
+only has the ability to change the number of knots
+in the spectral direction (i.e. the blaze function).
+If the spatial slit profile is not calculated, the
+blaze function will still be calculated using the
+'reduce flatfield' settings listed above.
+
+
+For Developers
+==============
+
+One of the ways the edge-finding algorithm is fooled is
+via chip defects, e.g. bad columns.  It is therefore
+valuable to mask any such known features with the
+bad pixel mask when one introduces a new instrument
+(or detector).
```

Comment:
Perhaps if there's any left/right edge boundaries identified in the first instance of edgearr, where the left edge is 1~2 pixels apart,  we can zero out those detections edge detections.

Reply:
This is a good suggestion.  I can't recall if one of each edge was generated.
I suspect not given how 'sharp' a bad column is..